ghproxy/ghmetrics/ghpaths.go

```diff
@@ -0,0 +1,295 @@
+package ghmetrics
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	reposRegex          = regexp.MustCompile(`^/repos/(?P<owner>[^/]*)/(?P<repo>[^/]*)(?P<rest>/.*)$`)
```

Comment:
I'm still not sure this is the most appropriate way to be parsing this. Maybe we can implement an actual descent parser and structure the data in a trie? We're just splitting by slashes and following down the chain to a static representation

Reply:
Example: https://play.golang.org/p/KWmT9hluIBE

Reply:
(could make the initialization simpler, too)